product_pricelist_supplierinfo/models/product_product.py

```diff
@@ -9,6 +9,17 @@
 class ProductProduct(models.Model):
     _inherit = 'product.product'
 
+    def _prepare_sellers(self, params):
+        """When we override min qty we want that _select_sellers gives us the
+        first possible seller for every other criteria ignoring the quantity. As
+        supplierinfos are sorted by min_qty descending, we want to revert such
+        order so we get the very first one, which is probably the one to go.
+        """
+        sellers = super()._prepare_sellers(params)
+        if self.env.context.get("override_min_qty"):
+            sellers = sellers.sorted("min_qty")
+        return sellers
```

Comment:
Maybe this code is not necessary as it exits the loop on the first match:
https://github.com/odoo/odoo/blob/5ab778227dfbc93b2ff415ebb0e40ce35953f13e/addons/product/models/product.py#L594
Luckily the default order is as expected:
https://github.com/odoo/odoo/blob/f5751e24be857d5be6969d85225677d7da5f0baa/addons/product/models/product.py#L740